### ISSUE TITLE
Patterns with hyperlink can't be autolink

### ISSUE THREAD

**Issue by shawn111** [mattermost/mattermost-plugin-autolink#26]
Hi,

I like this plugin, and just found this issue.

In the test cases and examples, we have below patterns.
```
 {
   "Pattern": "(https://mattermost.atlassian.net/browse/)(MM)(-)(?P<jira_id>\\d+)",
   "Template": "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)"
 },
```
It can pass the unittest.

However, it isn't working in the real services.
I guess the hyperlink is already be modified that cause the pattern not work.

**Comment by CrowdfoxMKL** [comment]
I have to second this - I tried the samples from the README, but to no avail. I even tested this further and found out, as soon as the pattern is to be compared to a actual link in Mattermost, it doesn't get modified. 
As a sample try these to pattern/template combos:
```
                {
                    "Pattern": "(https:test)",
                    "Template": "found"
                },
                {
                    "Pattern": "(https://test)",
                    "Template": "found"
                },
```
As you will see, the first pattern matches if you type the pattern im Mattermost, the second pattern does not match.

I'd really like to use this, so I can clean up the conversations of our developers a bit :)

**Comment by grubbins** [comment]
I also encountered this problem. It will affect anything that gets replaced with an `AutoLink` during markdown processing, which currently applies to the following URL schemes:
```golang
DefaultUrlSchemes = []string{"http", "https", "ftp", "mailto", "tel"}
```

The content of these is skipped by the plugin because it is no longer raw text. I believe I have a fix (PR on the way).

**Comment by grubbins** [comment]
The issue can also be demonstrated by adding suitable cases to `plugin_test.go` which applies the markdown parser (unlike `autolinker_test.go`).

**Comment by jespino** [comment]
Fixed in the PR #38
